api_design/DDS-1061-optimized_directory_structure_endpoint.md

```diff
@@ -0,0 +1,155 @@
+# DDS-1061 Optimized Directory Structure Endpoint
+
+## Deployment View
+
+status: submitted for review
+
+###### Deployment Requirements
+
+## Logical View
+
+The goal of this development epic is to produce a set of endpoints to be used in place of the `/projects/:id/children` and `/folders/:id/children` endpoints.
+
+#### Background
+
+The legacy children endpoints have proven to be suboptimal for the GCB clients needs. The GCB client needs a paginated endpoint to list ALL children of a project or folder. The current children endpoints meet the information needs for their use, but it currently takes on average 15 minutes to get the directory structure for a project or folder with many subfolders and files (similar to our ua_test sprawl project).
```

Comment:
The use case for the GCB client we have is uploading and downloading multiple files, usually entire projects. While the deliverable for such a use case might end up being a "paginated endpoint to list ALL children of a project or folder", I wouldn't say that's we're specifically asking for that.

Reply:
That is definitely a goal of this part of the feature development process, to get an exact spec of your needs. 

We will change the wording of the Background to `The GCB client needs an endpoint or set of endpoints to facilitate uploading and downloading of multiple files, usually entire projects`.

So far I have that we will not need to implement `name_contains` or `exclude_fields`. Your client will always ask for all children of a project or folder.

We will develop a new response document proposal and push to this branch to update the PR, and repeat the review.